Update retention function

TellAnAx opened the pull request:
closes issue #33 

**Additional changes**
- renamed function
- renamed function arguments

@Daamasi @Madhav-karthikeyan I am not yet content with the wording/naming. What we are calculating is the absolute mass change of a substance in the tissue of the animal. Retention is thus actually incorrect. What other term would come to your mind? Absolute change?

TellAnAx committed:
change m_start and m_end to ibw and fbw

TellAnAx committed:
rename retention into nr

TellAnAx committed:
update checks and rename nut_start and nut_end to ibn and fbn

TellAnAx committed:
debug function

TellAnAx committed:
update and expand tests

TellAnAx committed:
add examples

TellAnAx committed:
Merge branch 'cran' into update_retention

Daamasi commented on R/nr.R:
```diff
@@ -0,0 +1,67 @@
+#' # final mass fraction of N = 150 g/kg (15%)
+#' # -> 14 g N have been taken up.
+#' nr(ibw = 10, fbw = 100, ibn = 0.1, fbn = 0.15)
+#' 
```
Nice! We should get examples on most other functions, so it is easier for newcomers to have a grasp. Task for another update?

TellAnAx replied:
@Daamasi yes, even necessary for CRAN submission! ;) All functions must have examples provided.

Daamasi commented on R/nr.R:
```diff
@@ -0,0 +1,67 @@
+#' nutrient in the tissue of the experimental animal.
+#' @param fbn numeric between 0 and 1; final mass fraction of the target 
+#' nutrient in the tissue of the experimental animal.
+#' 
```
From a quick search (https://www.nature.com/articles/s43016-024-00932-z#Sec6) seems that ibn should be actually the nutrient present in the food, and fbn the same nutrient in the fish. 
Can't be sure for now and would have to take a deeper look into it. 
@TellAnAx can you check?

TellAnAx replied:
@Daamasi so basically it is 

```
retention = (fbw * fbn - ibw * ibn) / (m_feed * y_nutrient)
```

where y_nutrient is the mass fraction of the nutrient in the feed. Makes sense! I wonder how I came up with this messed up version 🤔 😅

TellAnAx replied:
addressed by commit 09f2c52

Daamasi commented on R/nr.R:
```diff
@@ -0,0 +1,67 @@
+#' # multiple fish
+#' nr(ibw = 10, fbw = c(93, 102, 99, 98, 101, 132), ibn = 0.1, fbn = 0.15)
+#' 
+#' 
```
Requires a reference. Maybe https://www.nature.com/articles/s43016-024-00932-z#Sec6, but there are surely foundational papers which should be more suited for this

TellAnAx replied:
There must be something more foundational in animal nutrition literature, but perhaps better this reference than none for now! 😬

TellAnAx replied:
addressed by commit 5b1a763

TellAnAx committed:
correct function, add additional function arguments and update checks and documentation

TellAnAx committed:
update tests

TellAnAx committed:
Merge branch 'update_retention' of https://github.com/TellAnAx/aquacultuR into update_retention

TellAnAx committed:
add reference: Willer et al. (2024)

TellAnAx commented:
@Daamasi suggested changes were addressed as requested! 

@SimaoCorreia please take a look at the update! I think the equation in the formulary also needs an update.

Daamasi commented on R/nr.R:
```diff
@@ -4,12 +4,15 @@
+#' of the experimental animal. The value must be between 0 and 1.
+#' @param fbn numeric; final mass fraction of the target nutrient in the tissue 
+#' of the experimental animal. The value must be between 0 and 1.
+#' @param feed numeric; mass of feed given
```
Maybe use fi for feed intake for the sake of consistency. Add nuances of feed intake and feed fed to the documentation on a later version?

TellAnAx replied:
now addressed by commit ba3589d

Daamasi commented on R/nr.R:
```diff
@@ -29,39 +32,51 @@
+               fbn,
+               feed,
+               nut_diet) {
 
```
If you change feed for fi in @params make sure you do it in the function args

TellAnAx replied:
now addressed by commit ba3589d

Daamasi commented on R/nr.R:
```diff
@@ -29,39 +32,51 @@
   # Calculations----
-  nr <- fbw * fbn - ibw * ibn
+  nr <- (fbw * fbn - ibw * ibn) / (feed * nut_diet)
   
```
This equation still doesn't make perfect sense to me, but I must say I haven't studied it enough! The reason is how can we get the initial nutrient fraction on the fish without killing it? We can have a sub-sample, but even then we would be loosing biological samples... Maybe we can have equally valuable info just by the ratio of what was "transfered" from the feed into the final animal. For instance, we gave x Kg of nutrient A through the feed, of all that A, y Kg of nutrient A ended up in the final animal body. 
Feel free to disregard this if it does not make sense.

Daamasi replied:
@TellAnAx have a look at this

TellAnAx replied:
@Daamasi You are right, the initial body composition needs to be determined by killing some of the fish initially. That's why you would usually have some "backup fish" and feed them with the same feed over some time before you do the initial stocking and eventually start the experiment.

TellAnAx committed:
change feed to fi and update examples

TellAnAx committed:
add additional information regarding body composition being required on dry matter basis to the documentation

TellAnAx commented:
@Daamasi I have changed `feed` to `fi` as suggested, and also corrected the examples (forgot to add `feed` and `nut_diet` there). See my comment regarding the initial and final body composition. 

Another important update is that the body composition data must be based on dry matter (just realized it while updating the examples). I mentioned this in the documentation (commit c9986de).
